lib/FrameManager.js

```diff
@@ -240,6 +240,20 @@ class Frame {
 
   /**
    * @param {string} selector
+   * @param {Function|string} pageFunction
+   * @param {!Array<*>} args
+   * @return {!Promise<(!Object|undefined)>}
+   */
+  async $$eval(selector, pageFunction, ...args) {
+    const arrayHandle = await this._context.evaluateHandle(selector => Array.from(document.querySelectorAll(selector)), selector);
+    args = [arrayHandle].concat(args);
```

Comment:
nit: Can avoid reassigning to args
`await this.evaluate(pageFunction, arrayHandle, ...args)`